mobile/lib/modules/album/services/album.service.dart

```diff
@@ -37,56 +110,51 @@ class AlbumService {
     Iterable<User> sharedUsers = const [],
   ]) async {
     try {
-      final dto = await _apiService.albumApi.createAlbum(
+      AlbumResponseDto? remote = await _apiService.albumApi.createAlbum(
         CreateAlbumDto(
           albumName: albumName,
           assetIds: assets.map((asset) => asset.remoteId!).toList(),
           sharedWithUserIds: sharedUsers.map((e) => e.id).toList(),
         ),
       );
-      return dto != null ? Album.remote(dto) : null;
+      if (remote != null) {
+        Album album = await Album.remote(remote);
+        await _db.writeTxn(() => _db.albums.store(album));
```

Comment:
What is the different of just using `_db.albums.store(album)` vs put it into the closure of `_db.writeTxn()`

Reply:
everything that changes (writes to) the Isar DB needs to be placed in an explicit write transaction.
If you forget it, Isar throws an error and aborts